crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -405,14 +405,24 @@ impl Analyzer<'_, '_> {
                             mark_var_as_truthy = true;
                             ty
                         })
-                        .map_err(|v| {
-                            match v.actual() {
-                                Error::CannotAssignToNonVariable { .. } => {
-                                    skip_right = true;
-                                }
-                                _ => {}
+                        .convert_err(|err| {
+                            skip_right = true;
+                            match err.actual() {
+                                Error::CannotAssignToNonVariable { .. } => match analyzer.scope.vars.get(&i.into()) {
+                                    Some(v) if v.kind == VarKind::Fn => Error::CannotAssignToFunction { span },
+                                    _ => err,
+                                },
+                                Error::NotVariable { .. } => match analyzer.type_of_raw_var(i, TypeOfMode::LValue) {
```

Comment:
This is not good for performance. The extra `type_of_var` call is wasting a lot of cycles here. I think `Error::NotVariable` should be modified to contain `ty` it got.

Reply:
Thanks! I fixed this as you said, but I'm not sure I did it right. Could you review my fix? [`8507c7b` (#192)](https://github.com/dudykr/stc/pull/192/commits/8507c7b3075a853907dceaf800e09fc90a0fddb7)

Reply:
LGTM! cc @kdy1